spec/system/homepage_spec.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+describe "Homepage query parameters" do
+  it "Retains single query parameters" do
+    path = "/?test_one=true"
+    visit(path)
+    expect(page).to have_current_path(path)
```

Comment:
Hmm I don't think this is actually testing much... cos the assertion is gonna pass immediately, without the Ember app even booting up.

If we wanna do it this way, then we need to add something like `expect(page).to have_css(".topic-list")` before asserting the path. That way, selenium will wait for Ember to boot and render before moving on.

Or alternatively, we could add an assertion for arbitrary params to the existing [homepage-test.js](https://github.com/discourse/discourse/blob/main/app/assets/javascripts/discourse/tests/acceptance/homepage-test.js) qunit test. There is already an assertion for `?order=`, so adding another for arbitrary params should just be a case of copy/pasting 🤞

Reply:
oh so true, great call